.github/workflows/lock.yaml

```diff
@@ -8,6 +8,9 @@ on:
 
 jobs:
   lock:
+    permissions:
+      issues: write
+      pull-requests: write
```

Comment:
rm? we don't need write access to pull-requests.

Reply:
Hi @godofredoc , [dessant/lock-threads](https://github.com/dessant/lock-threads) by default locks both PRs and issues. For PRs, it needs `pull-requests:write` permission. There is a `process-only` option to restrict it to only issues or PRs, and the default is both...

BTW, I liked how this workflow pins the action to a commit rather than a version. That is part of the [GitHub security guidance](https://docs.github.com/en/actions/learn-github-actions/security-hardening-for-github-actions), and I have been thinking of adding automation for that, but not sure if developers will adopt it. Seeing it in use here makes me think they might...

Reply:
Thanks @varunsh-coder for the information. The expected behavior for this repo is to process only issues. I sent a PR to update the config https://github.com/flutter/flutter/pull/90023/files.

Regarding pinning commit versions this is something we started tracking actively for all the flutter repositories.

Reply:
I have removed the `pull-requests:write` permission